src/cluster_legacy.c

```diff
@@ -3633,7 +3637,9 @@ void clusterSendPing(clusterLink *link, int type) {
      * to put inside the packet. */
     estlen = sizeof(clusterMsg) - sizeof(union clusterMsgData);
     estlen += (sizeof(clusterMsgDataGossip)*(wanted + pfail_wanted));
-    estlen += writePingExt(NULL, 0);
+    if (server.dbg_node_send_extensions && link->node && nodeSupportsExtensions(link->node)) {
+        estlen += writePingExt(NULL, 0);
+    }
```

Comment:
nit - this would read as if we only send extensions when the dbg flag is enabled. let's use the invert instead.

```suggestion
    if (!server.dbg_suppress_clustermsg_extensions && link->node && nodeSupportsExtensions(link->node)) {
        estlen += writePingExt(NULL, 0);
    }
```

Reply:
on a second thought, this dbg flag doesn't do anything because our receiver knows how to handle the extensions, now that we no longer disable extension handling logic using the dbg flag, which I think is the right thing to do.

I think a lot of the complexity of this fix comes from trying to create a TCL test for ad hoc version upgrade tests.  This is going to create headaches in the long run. I would suggest removing this debug flag and the TCL tests, and then validating the upgrade path *manually* for this fix. Let's defer discussion of the version upgrade test framework.